src/CoreWCF.Primitives/tests/DispatcherClient/DispatcherChannelFactory.cs

```diff
@@ -0,0 +1,153 @@
+    {
+        private IServiceProvider _serviceProvider;
+        private IDictionary<object, IServiceChannelDispatcher> _serviceChannelDispatchers = new Dictionary<object, IServiceChannelDispatcher>();
+        private object _lock = new object();
+
+        public DispatcherChannelFactory(Action<IServiceCollection> configureServices)
+        {
+            _serviceProvider = BuildServiceProvider(configureServices);
+        }
+
+        private IServiceProvider BuildServiceProvider(Action<IServiceCollection> configureServices)
+        {
+            var services = new ServiceCollection();
+            services.AddSingleton<DispatcherChannelFactory>(this);
+            services.AddScoped<DispatcherReplyChannel>();
+            services.AddServiceModelServices();
+            IServer server = new Helpers.MockServer();
+            services.AddSingleton(server);
+            services.AddSingleton(GetType(), this);
+            configureServices?.Invoke(services);
+            var serviceProvider = services.BuildServiceProvider();
+            var serverAddressesFeature = serviceProvider.GetRequiredService<IServerAddressesFeature>();
+            server.Features.Set(serverAddressesFeature);
+            return serviceProvider;
+        }
+
+        internal override TimeSpan SendTimeout { get; } = TimeSpan.FromSeconds(60);
+        protected override TimeSpan DefaultCloseTimeout => CloseTimeout;
+        protected override TimeSpan DefaultOpenTimeout => OpenTimeout;
+
+
+        internal override TimeSpan CloseTimeout => TimeSpan.FromSeconds(30);
+        internal override TimeSpan OpenTimeout => TimeSpan.FromSeconds(30);
+
+        public TChannel CreateChannel(EndpointAddress to, Uri via)
+        {
+            var servicesScopeFactory = _serviceProvider.GetRequiredService<IServiceScopeFactory>();
+            var serviceScope = servicesScopeFactory.CreateScope();
+
+            if (typeof(TChannel) == typeof(IRequestChannel))
+            {
+                var channel = new DispatcherRequestChannel(serviceScope.ServiceProvider, to, via);
+                _ = GetServiceChannelDispatcher(channel);
+                return (TChannel)(object)channel;
+            }
+
+            if (typeof(TChannel) == typeof(IRequestSessionChannel))
+            {
+                var channel = new DispatcherRequestSessionChannel(serviceScope.ServiceProvider, to, via);
+                _ = GetServiceChannelDispatcher(channel);
+                return (TChannel)(object)channel;
+            }
+
+            //if (typeof(TChannel) == typeof(IDuplexSessionChannel))
+            //{
+            //    return (TChannel)(object)new DispatcherDuplexSessionChannel((DispatcherChannelFactory<IDuplexSessionChannel>)(object)this);
+            //}
+
+            throw new InvalidOperationException("Only support IRequestChannel and IDuplexSessionChannel");
```

Comment:
It should be "Only support IRequestChannel and IRequestSessionChannel", not "Only support IRequestChannel and IDuplexSessionChannel"